warehouse/templates/manage/release.html

```diff
@@ -142,7 +142,17 @@ <h3>{% trans %}Yank release{% endtrans %}</h3>
       For more information, see <a href="{{ href }}">PEP 592</a>.
     {% endtrans %}
     </p>
-    {{ confirm_button(gettext("Yank release"), gettext("Version"), "yank_version", release.version, modifier="--primary", warning=False) }}
+
+    {% set extra_description %}
+      <p>
+        {% trans project_name=project.name, version=release.version %}
+          You may provide a reason for yanking this release, which will be displayed by pip to users who install <code>{{ project_name }}=={{ version }}</code>.
+        {% endtrans %}
+      </p>
+      <label for="yanked_reason">{% trans %}Reason (optional){% endtrans %}</label>
+      <input name="yanked_reason" id="yanked_reason" type="text" autocomplete="off" autocapitalize="off" spellcheck="false">
+    {% endset %}
+    {{ confirm_button(gettext("Yank release"), gettext("Version"), "yank_version", release.version, modifier="--primary", warning=False, extra_description=extra_description) }}
```

Comment:
Two things I don't think I feel great about here:

1) `extra_description` probably isn't meant for this, but there wasn't another existing way to tie into `confirmation_button`/`confirmation_modal` and I wasn't sure I should introduce a whole new block for this singular use case.
2) The [duplication of this in `releases.html`](https://github.com/pypa/warehouse/pull/7916/files#diff-aa165515152660253988628f2a9af2caR118-R127) doesn't feel good so I'd need a suggestion about an approach befitting this code base to, say, put it in a shared location both templates could use.

Reply:
The `confirm_button`/`confirm_modal` pattern is generally getting a bit unwieldy, and we probably should think about ways to improve it, but I think this is probably fine as is for now.